datameta/models/db.py

```diff
@@ -196,15 +196,19 @@ class MetaDatumRecord(Base):
 class MetaDataSet(Base):
     """A MetaDataSet represents all metadata associated with *one* record"""
     __tablename__  = 'metadatasets'
-    id             = Column(Integer, primary_key=True)
-    site_id        = Column(String(50), unique=True, nullable=False, index=True)
-    uuid           = Column(UUID(as_uuid=True), unique=True, default=uuid.uuid4, nullable=False)
-    user_id        = Column(Integer, ForeignKey('users.id'), nullable=False)
-    submission_id  = Column(Integer, ForeignKey('submissions.id'), nullable=True)
+    id               = Column(Integer, primary_key=True)
+    site_id          = Column(String(50), unique=True, nullable=False, index=True)
+    uuid             = Column(UUID(as_uuid=True), unique=True, default=uuid.uuid4, nullable=False)
+    user_id          = Column(Integer, ForeignKey('users.id'), nullable=False)
+    submission_id    = Column(Integer, ForeignKey('submissions.id'), nullable=True)
+    is_deprecated    = Column(Boolean, default=False)
+    deprecated_label = Column(String, nullable=True)
+    replaced_by      = Column(Integer, ForeignKey('metadatasets.id'), nullable=True)
     # Relationships
     user             = relationship('User', back_populates='metadatasets')
     submission       = relationship('Submission', back_populates='metadatasets')
     metadatumrecords = relationship('MetaDatumRecord', back_populates='metadataset')
+    replacement_mset = relationship('MetaDataSet')
```

Comment:
Apparently sqlalchemy would default this to be a one-to-many relationship, i.e. as this is written down I think `replacement_mset` would in fact be a list of metadatasets, containing all the msets that have this mset in `replaced_by`. We should have both relationships anyway. My suggestion:

* rename the table column attribute `replaced_by` to be called `replaced_by_id`
* Provide two relationship attributes `replaced_by` (single element) and `replaces` (list) to model both directions

I haven't looked into the details, but this describes the pattern: https://docs.sqlalchemy.org/en/14/orm/self_referential.html

Reply:
updated according to SQLalchemy adjacency list pattern